library/core/src/time.rs

```diff
@@ -672,11 +672,8 @@ impl Duration {
             let mut nanos = self.nanos.as_inner() + rhs.nanos.as_inner();
             if nanos >= NANOS_PER_SEC {
                 nanos -= NANOS_PER_SEC;
-                if let Some(new_secs) = secs.checked_add(1) {
-                    secs = new_secs;
-                } else {
-                    return None;
-                }
+                let Some(new_secs) = secs.checked_add(1) else { return None };
+                secs = new_secs;
```

Comment:
I feel it is much harder to read the control flow in this case because we have hidden the `return` off to the right.